# Bioinformatics Algorithms

This repository contains a collection of bioinformatics algorithms developed as part of the course 
"Biology Meets Programming: Bioinformatics for Beginners" offered by the University of California, San Diego.

## Course Information

- **Course Name:** Biology Meets Programming: Bioinformatics for Beginners
- **Institution:** University of California, San Diego
- **Course Link:** [Course Page]((https://www.coursera.org/learn/bioinformatics/home/))

## Algorithms

### deamination.py

**Background:**

The `deamination.py` script is designed to handle deamination events in DNA sequences. 
Deamination is a biological process where the amino group of a base is removed or converted, 
often leading to mutations. This script provides functions to detect and analyze deamination events, 
which are critical for understanding DNA damage and repair mechanisms.

### hamming_distance.py

**Background:**

The `hamming_distance.py` script calculates the Hamming distance between two sequences of equal length. 
The Hamming distance measures the number of differing symbols between two sequences, 
making it a fundamental metric in genomics and sequence analysis.

### kmer_motif.py

**Background:**

The `kmer_motif.py` script deals with k-mers and motifs. 
It provides functions for identifying k-mers in sequences and analyzing sequence motifs. 
Understanding k-mers and motifs is essential for various applications, 
including DNA sequence alignment and regulatory element identification.

<img width="743" alt="Screenshot 2023-12-19 at 12 56 16 PM" src="https://github.com/katebarouch/most_common_kmer/assets/131073205/03ebb6ab-1d0d-44d9-bab4-87107048d9de">

### most_frequent_kmer.py

**Background:**

The `most_frequent_kmer.py` script identifies the most frequent k-mers in a given DNA sequence. 
It plays a vital role in finding significant patterns or motifs in biological data, 
which can be useful for gene expression analysis and transcription factor binding site prediction.

### reverse_complement_dna.py

**Background:**

The `reverse_complement_dna.py` script computes the reverse complement of a DNA sequence. 
The reverse complement is essential for tasks such as primer design, sequence alignment, 
and understanding the complementary base pairing in DNA molecules.

